controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -795,6 +796,63 @@ func TestRebootWithSuffixedAnnotation(t *testing.T) {
 	)
 }
 
+// TestRebootWithServicing tests full reboot cycle with suffixless
+// annotation and servicing.
+func TestRebootWithServicing(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = make(map[string]string)
+	host.Annotations[metal3api.RebootAnnotationPrefix] = ""
+	host.Status.PoweredOn = true
+	host.Status.Provisioning.State = metal3api.StateProvisioned
+	host.Spec.Online = true
+	host.Spec.Image = &metal3api.Image{URL: "foo", Checksum: "123"}
+	host.Spec.Image.URL = "foo"
+	host.Spec.Firmware = &metal3api.FirmwareConfig{
+		VirtualizationEnabled: ptr.To(true),
+	}
+	host.Status.Provisioning.Image.URL = "foo"
+
```

Comment:
A policy object must be created here, otherwise the test will never pass.

Reply:
Agreed, we just haven't figured out how to pass it in yet. The comment below suggests a promising path. In terms of creating the HUP object, do I need to do something similar to this: https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/baremetalhost_controller_test.go#L2319-L2356 ?

Reply:
Yes. This test is actually very similar to what you need: it also creates another object in addition to the BMH.

Reply:
thank you Dmitry, will try this later tonight or tomorrow